corehq/apps/userreports/tests/test_utils.py

```diff
@@ -95,3 +100,39 @@ def _check(doc_type, expected):
         (_check, 'DataSource-Deleted', 'DataSource'),
         (_check, 'DataSource-Deleted-Deleted', 'DataSource'),
     ]
+
+
+class TestGetDomainForUCRTableName(SimpleTestCase):
+
+    def test_modern_prefix(self):
+        domain = get_domain_url_slug('test')
```

Comment:
Hmm I don't like the idea of adding an additional assertion for `get_domain_url_slug` because that isn't what we are testing. The way I see it the assertions at the end of these tests (e.g., `self.assertEqual(result, 'test-u4500')`) should address your concern about domains being what we expect, whereas changing that assertion to `self.assertEqual(result, domain)` would obfuscate what the expected outcome is. Open to changing if something doesn't sit well with you though.

Reply:
I understand what you're saying, but as the test is written, if it failed it would be unclear if `get_domain_url_slug` returned the wrong thing or if `get_domain_for_ucr_table_name` was the culprit. Maybe `get_domain_url_slug` should be taken out of the tests, and they should all be written like the test in https://github.com/dimagi/commcare-hq/commit/db064096ba1fa69e6af19751aa3e519726955bd9?

It does seem important to verify that `get_domain_url_slug` converts underscores to something different, since `get_domain_for_ucr_table_name` depends on domain names never containing underscores.

Reply:
Ah that is fair. I've added unit tests for `get_domain_url_slug` which confirm that names are converted correctly, and have removed the use of that method from the `get_domain_for_ucr_table_name` tests. See c710977c5e80199c516a0815a71466033e53185d